## Reading, writing and displaying images with OpenCV

In [7]:
import cv2
import numpy as np

In [8]:
# Load an image using 'imread' specifying the path to image
input = cv2.imread('./images/input.jpg')

# To display our image variable, we use 'imshow'
# The first parameter will be title shown on image window
# The second parameter is the image varialbe
cv2.imshow('Hello World', input)

# 'waitKey' allows us to input information when a image window is open
# By leaving it blank it just waits for anykey to be pressed before 
# continuing. By placing numbers (except 0), we can specify a delay for
# how long you keep the window open (time is in milliseconds here)
cv2.waitKey()

# This closes all open windows 
# Failure to place this will cause your program to hang
cv2.destroyAllWindows()

In [9]:
input = cv2.imread('./images/input.jpg')

cv2.imshow('Hello World', input)
cv2.waitKey(0)
cv2.destroyAllWindows()

### Storing images

In [10]:
print(input.shape)

(830, 1245, 3)


#### Shape gives the dimensions of the image array

The 2D dimensions are 830 pixels in high bv 1245 pixels wide.
The '3' means that there are 3 other components (RGB) that make up this image.

In [11]:
# print each dimension of the image

print('Height of Image:', int(input.shape[0]), 'pixels')
print('Width of Image: ', int(input.shape[1]), 'pixels')

Height of Image: 830 pixels
Width of Image:  1245 pixels


### saving images we edit in OpenCV

In [12]:
# Simply use 'imwrite' specificing the file name and the image to be saved
cv2.imwrite('output.jpg', input)
cv2.imwrite('output.png', input)

True